File: Iteration2/code/CSE3063GRP12/src/test/TranscriptRepositoryTest.java
package test;
import static org.junit.Assert.assertEquals;
import java.io.IOException;
import org.junit.Test;

import core.exceptions.UserNotFoundException;
import core.models.concretes.Transcript;
import core.repositories.TranscriptRepository;

public class TranscriptRepositoryTest {

    @Test()
    public void testStudentCurrentSemester() throws IOException, UserNotFoundException {
        TranscriptRepository transcriptRepository = new TranscriptRepository();
        Transcript transcript = transcriptRepository.getTranscript("o150121534");
        assertEquals(transcript.getCurrentSemester(), 1);
    }

    @Test(expected = UserNotFoundException.class)
    public void testNotFoundTranscript() throws IOException, UserNotFoundException {
        TranscriptRepository transcriptRepository = new TranscriptRepository();
        transcriptRepository.getTranscript("o1501215");
    }

}

File: Iteration3/code/CSE3063GRP12/src/TranscriptRepositoryTest.py
import unittest
from core.repositories.TranscriptRepository import TranscriptRepository
from core.exceptions.UserNotFoundException import UserNotFoundException

class TranscriptRepositoryTest(unittest.TestCase):

    def test_student_current_semester(self):
        transcript_repository = TranscriptRepository()
        transcript = transcript_repository.get_transcript("o150120060")
        self.assertEqual(transcript.currentSemester, 6)

    def test_not_found_transcript(self):
        transcript_repository = TranscriptRepository()
        transcript = transcript_repository.get_transcript("o1501asdasd20060")
        self.assertIsNone(transcript)

if __name__ == '__main__':
    unittest.main()